####################################################################################################################################
####################################################################################################################################
############################################### DRONE AND TRAJECTORY PARAMETERS ####################################################
####################################################################################################################################
####################################################################################################################################

vxAmpl: 0.0    
vyAmpl: 1.0    
vzAmpl: 0.0    
angAmpl: 0.0   
f: 0.008   

# threshold: [0.001, 0.001, 0.001, 0.0875]
threshold: [0.1, 0.1, 0.1, 0.2]

inputRange: [1.0, 1.0, 1.0, 0.5]

####################################################################################################################################
##################################################### CONTROLLER SELECT ############################################################
####################################################################################################################################

#"FL" = Feedback Linearization - Bypassed ident
#"PID"          = PID
#"RLQR"         = Robust LQR
#"SLQR"         = Standard LQR
#"RecursiveLQR" = Recursive LQR
#"Hinfinite"    = H-Infinite Controller

controlSelect: "RLQR"

####################################################################################################################################
####################################################### SENSOR SELECT ##############################################################
####################################################################################################################################

#"IMU" = IMU
#"VICON"  = VICON
#"ORBSLAM"  = ORBSLAM

sensorSelect : "VICON"

####################################################################################################################################
################################################# PARAMETER IDENTIFICATION #########################################################
####################################################################################################################################

# /*K for IMU*/ DRONE 1 CROB
Kimu: [1.7420, 0.9402, 1.5441, 0.8963, 3.3489, 3.2947, 6.5121, 3.9219]
# /*K for IMU*/ DRONE PRETO ROBERTO (NAO TESTADO)
# Kimu: [1.2327, 0.0943, 1.0457, 0.1490, 3.3489, 3.2947, 6.5121, 3.9219]


# /*K for VICON + Drone 1 => Sep/18/2019*/
# Kvicon: [3.6976, 0.1832, 3.8513, 0.3573, 2.4596, 2.1683, 4.4168, 2.7023]
# /*K for VICON*/ DRONE PRETO ROBERTO
# Kvicon: [4.0527, -0.1135, 4.1320, 0.0836 , 2.4596, 2.1683, 4.4168, 2.7023] #FUNCIONANDO..Z MEIO CAPENGA
# Kvicon: [4.0527, -0.1135, 4.1320, 0.0836 , 5.1451, 3.2975, 4.4168, 2.7023] #FUNCIONANDO..Z MELHOR

# K para Vicon a 20Hz - Drone 4 Branco

Kvicon: [3.8166, 0.0525, 3.9937, 0.2669, 2.5130, 2.5079, 7.3590, 2.9258]



# /* K for VICON 15/10/2018 - Using updated ModelIdentification.m*/
#K: [1.7420, 0.9402, 1.5441, 0.8963, 3.3489, 3.2947, 6.5121, 3.9219]
#K: [4.5752, 0.1149, 5.2296, 0.2364, 4.4206, 3.1341, 5.9289, -0.3868] #Valor que tava indo!
# K: [4.3321, 0.2333, 4.1220, 0.4145, 4.4206, 3.1341, 5.9289, -0.3868]

K: [3.2657, 1.5034, 2.9334, 1.1048, 3.3489, 3.2947, 6.5121, 3.9219]   #IDENTIFICADO COM MODELO DISCRETIZADO VIA EKF (EXCETO GAMMAS 7 E 8)


####################################################################################################################################
#################################################### CONTROLLER GAINS ##############################################################
####################################################################################################################################

# The following PID parameters worked fine on a 100Hz control loop

# Kp: [-0.70, -0.70, -2.75, -1.2]
# Kd: [-0.75, -0.75, -0.75, -3.35]


# Kp: [-0.5, -0.5, -0.5, -1.2] # before 12/04/2018


# Kd: [-1.175, -1.175, -1.175, -3.35] # before 12/04/2018
# Kd: [-1.175, -1.175, -1.175, -0.0]

# Ki: [-0.0, -0.0, -0.0, -0.0] # before 12/04/2018
Ki: [-0.01, -0.01, -0.01, -0.01]


# ------------------ FL control parameters ------------------  

KpFL: [1.33, 1.40, 3.50, 1.75]

KdFL: [1.50, 1.50, 2.00, 2.00]



# Os parâmetros PID abaixo funcionaram bem na Vicon, ainda melhor, para 100Hz.

# Kp: [-0.5, -0.5, -0.5, -1.2]

# Kd: [-0.5, -0.5, -0.5, -0.8]

# Ki: [0.0, 0.0, 0.0, 0.0]

# Kp: [-0.315, -0.002, -0.002, -0.002]

# Kd: [-2.4, -0.0, -0.0, -0.0]

# Ki: [-0.06, 0.0, 0.0, 0.0]

# Parâmetros para Tuning do Feedbback Linearization.

# Kp: [-1.33, -1.40, -3.50, 0]

# Kd: [-1.50, -1.50, -2.00, 0]

Kp: [0.000021, 0, 0, 0]

Kd: [0, 0, 0, 0]

# Kp: [0.05, 0.05, 0.05, 1.2]

# Kd: [0.05, 0.05, 0.05, 0.8]


# Kp: [-5.0, -5.0, -0.04, -0.09]

# Kd: [-1.0, -1.0, -0.06, -0.095]


####################################################################################################################################
##################################################### TRAJECTORY SELECT ############################################################
####################################################################################################################################

# "eightShape"      = 8-shape  
# "circleXY"        = circle in X-Y axis only 
# "circleZXY"       = circle in X-Y-Z axis. 
# "straightLine"    = point-to-point (x=y=amplitude)
# "wayPoint"        = set point without acc and vel desired
# "ident"           = Identification Flight = circleXY (30s) + sinus Z(30s) + sinus Yaw (remaining time)


# comment
trajectory: "circleXY"

autoMode: 0

velMed: 0.15

amplitude: 0.5

# In case we select "straightLine", poseDesired is defined as: [Xd[m], Yd[m], Zd[m], Yawd[rad], Time[s]]
poseDesired: [0, 0 , 0.6, 0, 5]

####################################################################################################################################
################################################## EKF ONLINE ESTIMATION ###########################################################
############################################### (FOR FUTURE VERSIONS ONLY) #########################################################
####################################################################################################################################

# isEKFonline enables (true = 1) or disables (false = 0) online EKF estimation for deltaM and deltaN
isEKFonline: 0

updateRateEKF: 5


####################################################################################################################################
################################################## CENTRALIZED ESTIMATION ##########################################################
####################################################################################################################################

# Flag that enables the Centralized Moving Horizon Estimation (CMHE). 0 = Disabled; 1 = Enabled
isCMHEenabled: 1

# Defines the number of agents to be considered
nOfAgents: 1

# Defines the buffer Size
bfSize: 5

# Defines dropout Probability on incoming message (VICON): A double between 0 and 1
dropProbability: 0.1

# Defines maximum number consecutive packet dropouts in a row (works for one agent only)
nLossMax: 2